### DISCUSSION TITLE
Add the Deno template?

### DISCUSSION THREAD

**Discussion by jcbhmr** [devcontainers-community/templates#24] [Ideas]
After #17 decided against adding vue (too similar to plain js), the next one on the list is a deno template. https://deno.com/runtime

![image](https://user-images.githubusercontent.com/61068799/235492152-6f9076a0-46cb-46c5-8994-c3c6fd3ea38e.png)

https://github.com/orgs/microsoft/packages?q=vscode-dev-containers&sort_by=downloads_desc

the vscode-dev-containers one is the only "deno" result via Ctrl+F in the template index: https://containers.dev/templates

so...

the original: https://github.com/microsoft/vscode-dev-containers/tree/main/containers/deno
last update was 7 months ago.
original contributors (unsure if they were maintainers?) were: @anthonychu, @kitsonk

this DISCUSSION is intended to be an outreach to @anthonychu, @kitsonk to see if they are interested in moving from the "Deprecated (No longer maintained)" vscode-dev-containers to this @devcontainers-community

so, on that note... 👋 Hi there @anthonychu, @kitsonk! Would you be interested in a "passing the torch" of https://github.com/microsoft/vscode-dev-containers/tree/main/containers/deno to @devcontainers-community? Would you prefer to be added as a member of this organization? Added as a write-access contributor to this repository? Should the Deno template be left as-is? ❤️ Could an official Deno devcontainer template repository that could be pursued instead?

related previous jupyter datascience addition discussion/issue so you can see a previous example: https://github.com/devcontainers-community/templates/issues/2

---

@eitsupi is this good idea in keeping in-scope and with the spirit and core values of @devcontainers-community ?? looking for a second validating opinion

> The [@devcontainers-community](https://github.com/devcontainers-community)'s mission is to be a centralized source for well-maintained features that are of high-quality similar to those of the official [@devcontainers](https://github.com/devcontainers) organization, but for additional popular tools and languages.
>
> We want to avoid atrophy by investing into the community-based approach to maintenance that has been successful in other communtiy orgainzations like [@asdf-community](https://github.com/asdf-community). This means making it as easy as possible to contribute. 🤝
>
> This is, however, a fallback option for languages and tools that do not have official awesome-tool/devcontainer-features or related repositories to self-maintain within that community. If you are an individual who is part of a popular tool, consider [authoring your own feature repository](https://github.com/devcontainers/feature-starter#readme)! ❤️

https://github.com/devcontainers-community/.github#readme

**Comment by jcbhmr** [comment]
There's a _potential_ branch at https://github.com/devcontainers-community/templates/tree/deno

**Comment by eitsupi** [comment]
I agree with adding Deno.

Perhaps we also need a Feature to install Deno....... (Deno does not have an official arm64 binary, so the Feature may need an option to install an unofficial arm64 binary)

**Reply by jcbhmr** [reply]
@devcontainers-contrib does actually already have a deno feature! in fact, there's two! it's just that the way its gone about is... not not the most _efficient_ (install asdf just to install deno?) or "official"ish imo. but that's my opinion.

![image](https://user-images.githubusercontent.com/61068799/235495495-18e19a06-cd2a-4af0-a815-d5e5de4985c5.png)

we **could** use that or maintain our own that just relegates to installing curl, unzip, and then `curl -fsSL https://deno.land/x/install/install.sh | sh` depending on whether or not that's worth the effort to make.

> so the Feature may need an option to install an unofficial arm64 binary)

this would be a **very valuable** addon/expansion/option! 🥇 i dont think the current two features offer that

**Reply by eitsupi** [reply]
I do not know if the official installation scripts match the demand to install a specific version.
(I've heard that Deno itself has the ability to switch versions, so this may not actually be a problem.)

**Reply by jcbhmr** [reply]
yup! you can switch version on-the-fly via the `deno upgrade` command:

![image](https://user-images.githubusercontent.com/61068799/235499392-a272b73f-2066-4603-84da-15d3de47afd4.png)

https://deno.com/manual@v1.33.1/getting_started/installation#updating

looks like you can also set the version via a script arg `$1` of the installer

```sh
curl -fsSL https://deno.land/x/install/install.sh | sh -s v1.24.1
```

**Reply by eitsupi** [reply]
> looks like you can also set the version via a script arg `$1` of the installer
> 
> ```shell
> curl -fsSL https://deno.land/x/install/install.sh | sh -s v1.24.1
> ```

Sounds good!
The Julia Feature does the same thing.
https://github.com/JuliaLang/devcontainer-features/blob/45c7a4d2c7ae4f451ddc1692705e15ce931c43dc/src/julia/install.sh#L52-L58

As for arm64 support, I think we need to write a complete installation script for now.

**Reply by jcbhmr** [reply]
continued in https://github.com/devcontainers-community/features/discussions/18

**Comment by anthonychu** [comment]
> this DISCUSSION is intended to be an outreach to @anthonychu, @kitsonk to see if they are interested in moving from the "Deprecated (No longer maintained)" vscode-dev-containers to this https://github.com/devcontainers-community

I think I helped with this initially but haven't done much to help maintain it since, will defer to @kitsonk.
